hello_network/lib/hello_network.ex

```diff
@@ -1,18 +1,61 @@
 defmodule HelloNetwork do
+  @moduledoc """
+  Example of setting up Networking on a Nerves device.
+  """
 
-  alias Nerves.Networking
+  require Logger
 
-  @interface :eth0
+  alias Nerves.{Network}
 
+  @interface "eth0"
+  @settings [ipv4_address_method: :dhcp]
+
+  @doc "Main entry point into the program. This is an OTP callback."
   def start(_type, _args) do
-    unless :os.type == {:unix, :darwin} do     # don't start networking unless we're on nerves
-      {:ok, _} = Networking.setup @interface
-    end
-    {:ok, self()}
+    GenServer.start_link(__MODULE__, {@interface, @settings}, [name: __MODULE__])
   end
 
-  @doc "Attempts to perform a DNS lookup to test connectivity."
+  @doc "Are we connected to the internet?"
+  def connected?, do: GenServer.call(__MODULE__, :connected?)
+
+  @doc "Returns the current ip address"
+  def ip_addr, do: GenServer.call(__MODULE__, :ip_addr)
+
+  @doc """
+  Attempts to perform a DNS lookup to test connectivity.
+
+  ## Examples
+
+    iex> HelloNetwork.test_dns()
+    {:ok,
+     {:hostent, 'nerves-project.org', [], :inet, 4,
+      [{192, 30, 252, 154}, {192, 30, 252, 153}]}}
+  """
   def test_dns(hostname \\ 'nerves-project.org') do
     :inet_res.gethostbyname(hostname)
   end
+
+  ## GenServer callbacks
+
+  def init({interface, settings}) do
+    Network.setup(interface, settings)
+
+    SystemRegistry.register
+    {:ok, %{ interface: interface, ip_address: nil, connected: false }}
+  end
+
+  def handle_info({:system_registry, :global, registry}, state) do
+    ip = get_in registry, [:state, :network_interface, state.interface, :ipv4_address]
+    if ip != state.ip_address do
+      Logger.info "IP ADDRESS CHANGED: #{ip}"
+    end
+
+    connected = match?({:ok, {:hostent, 'nerves-project.org', [], :inet, 4, _}}, test_dns())
+    {:noreply, %{state | ip_address: ip, connected: connected || false}}
+  end
+
+  def handle_info(_, state), do: {:noreply, state}
```

Comment:
If you implement one `handle_info` you have to ignore all other things. if you don't implement a `handle_info` it will ignore all random info iirc.